src/config.rs

```diff
@@ -1329,20 +1328,14 @@ pub mod config_file {
         Ok(toml::from_str(&read_to_string(file)?)?)
     }
 
-    fn read_file<F: FnOnce() -> Option<PathBuf>>(
-        dir: F,
-        file: &str,
-    ) -> anyhow::Result<Option<ConfigFile>> {
-        if let Some(mut path) = dir() {
-            path.push(file);
+    fn read_file<P: AsRef<Path>>(dir: P) -> anyhow::Result<Option<ConfigFile>> {
```

Comment:
I added a `read_files` function to check for both the non-hidden and hidden files to keep `read_file` simple.